docs/concepts/configuration/taint-and-toleration.md

```diff
@@ -0,0 +1,257 @@
+hardware off of those nodes, thus leaving room for later-arriving pods that do need the
+specialized hardware. This can be done by tainting the nodes that have the specialized
+hardware (e.g. `kubectl taint nodes nodename special=true:NoSchedule` or
+`kubectl taint nodes nodename special=true:PreferNoSchedule`) and adding a corresponding
+toleration to pods that use the special hardware. As in the dedicated nodes use case,
+it is probably easiest to apply the tolerations using a custom
+[admission controller](/docs/admin/admission-controllers/)).
+For example, the admission controller could use
+some characteristic(s) of the pod to determine that the pod should be allowed to use
+the special nodes and hence the admission controller should add the toleration.
+To ensure that the pods that need
+the special hardware *only* schedule onto the nodes that have the special hardware, you will need some
+additional mechanism, e.g. you could represent the special resource using
+[opaque integer resources](/docs/concepts/configuration/manage-compute-resources-container/#opaque-integer-resources-alpha-feature)
+and request it as a resource in the PodSpec, or you could label the nodes that have
+the special hardware and use node affinity on the pods that need the hardware.
+
+* **Taint based Evictions (alpha feature)**: A per-pod-configurable eviction behavior
+when there are node problems, which is described in the next section.
+
+## Taint based Evictions
+
+Earlier we mentioned the `NoExecute` taint effect, which affects pods that are already
+running on the node as follows
+
+ * pods that do not tolerate the taint are evicted immediately
+ * pods that tolerate the taint without specifying `tolerationSeconds` in
+   their toleration specification remain bound forever
+ * pods that tolerate the taint with a specified `tolerationSeconds` remain
+   bound for the specified amount of time
+
+The above behavior is a beta feature. In addition, Kubernetes 1.6 has alpha
+support for representing node problems. In other words, the node controller
+automatically taints a node when certain condition is true. The builtin taints
+currently include:
+
+ * `node.alpha.kubernetes.io/notReady`: Node is not ready. This corresponds to
+   the NodeCondition `Ready` being "`False`".
+ * `node.alpha.kubernetes.io/unreachable`: Node is unreachable from the node
+   controller. This corresponds to the NodeCondition `Ready` being "`Unknown`".
+ * `node.kubernetes.io/outOfDisk`: Node becomes out of disk.
+ * `node.kubernetes.io/memoryPressure`: Node has memory pressure.
+ * `node.kubernetes.io/diskPressure`: Node has disk pressure.
+ * `node.kubernetes.io/networkUnavailable`: Node's network is unavailable.
+ * `node.cloudprovider.kubernetes.io/uninitialized`: When kubelet is started
+   with "external" cloud provider, it sets this taint on a node to mark it
+   as unusable. When a controller from the cloud-controller-manager initializes
+   this node, kubelet removes this taint.
```

Comment:
To Reviewers: Line 190-205 are new contents, mainly enumerating the various built-in taints supported now.